src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,230 @@
+ */
+class CsvToHttp extends ApiHandler {
+  /**
+   * Constructor for CsvToHttp
+   * @constructor
+   * @param {Object} applicationParameters - The application parameters
+   * @param {Engine} engine - The Engine
+   * @return {void}
+   */
+  constructor(applicationParameters, engine) {
+    super(applicationParameters, engine)
+
+    const { applicativeHostUrl, requestMethod, authentication, csvDelimiter, mapping, proxy = null } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {}
+    this.request.host = applicativeHostUrl
+    this.request.method = requestMethod
+    this.request.authentication = authentication
+
+    this.mapping = mapping || {}
+    this.mapping.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.httpBody = []
+    this.canHandleFiles = true
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format, file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    await this.parseAndMapp(filePath)
+
+    if (this.httpBody.length !== 0) {
+      await this.sendData()
+    }
+
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - Path of the file to parse
+   * @return {Promise} - Promise
+   */
+  parseAndMapp(filePath) {
+    try {
+      // Initialize array wich will be filled at each step
+      const csvDataParsed = []
+      const csvFile = fs.createReadStream(filePath)
+      // Parse the csv file into a json object
+      Papa.parse(csvFile, {
+        delimiter: this.mapping.csvDelimiter,
+        header: true,
+        dynamicTyping: true,
+        error: (error) => {
+          this.logger.error(error)
+          return ApiHandler.STATUS.LOGIC_ERROR
+        },
+        step: (step) => {
+          csvDataParsed.push(step.data)
+        },
+        complete: () => {
+          // The csv parsing is a success so we begin to map it
+          // Initialize the body to an empty array
+          this.httpBody = []
+          this.convertToHttpBody(csvDataParsed)
+        },
+      })
+    } catch (error) {
+      this.logger.error(error)
+    }
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Promise} - The send status
+   */
+  convertToHttpBody(jsonObject) {
+    try {
+      if (jsonObject.length > 0) {
+        jsonObject.forEach((jsonObj) => {
+          const response = this.convertCSVRowIntoHttpBody(jsonObj)
+          // Test the result of the mapping/convertion before add it in the httpBody
+          if (response) {
+            this.httpBody.push(response)
```

Comment:
For me a global variable has the good scope but it's not the good solution because it main contains sensitive information.

Moreover, a class variable is so much easier in the comprehension of the code (same variable from the beginning to the end),  

If you really want a local variable I can implement it but in the optic to parse other type of data, it would be possible to do both at the same time ( httpBodyFromCsv and httpBodyFromJson...). However you may implement 2 different methods and different local variable.